### PR TITLE
Change defeat conditions

### PR DESCRIPTION
**Pull request by SomeTroglodyte**
- Implements defeat conditions as per #4707, ravignir's clarification
- Fixes a typo - thus all language files touched
- Fixes 2^31-1 showing as turns-to-tech when science per turn is zero (possible when testing item 1) - why that division by zero doesn't crash is a mystery
- Fixes crash when player loses (regression from introducing victory type as unique parameter)

Holler if you'd rather not do the changed defeat logic, I would totally understand - I'll isolate the fixes.

### CODE REVIEW

**Comment by SomeTroglodyte** [comment]
Wow, not AS stealing blanks but `lateinit property name has not been initialized` - each time I forget to run the UT locally it bites back. Gimme a quarter.

**Comment by SomeTroglodyte** [comment]
Turns out the mock civInfos in the diplomacy tests were a bit fragile and stumbled over isBarbarian() crashing - I hope this approach is OK, adding a unit test just to make sure the prerequisites for the main unit tests are met may not be a good pattern. At least this way such dependencies are clearly documented...

File: core/src/com/unciv/logic/civilization/TechManager.kt [hunk 1]
```diff
@@ -113,8 +113,10 @@ class TechManager {
     
     fun remainingScienceToTech(techName: String) = costOfTech(techName) - researchOfTech(techName)
 
-    fun turnsToTech(techName: String): String {
-        return if (civInfo.cities.isEmpty()) "∞" else max(1, ceil(remainingScienceToTech(techName).toDouble() / civInfo.statsForNextTurn.science).toInt()).toString()
+    fun turnsToTech(techName: String) = when {
+        civInfo.cities.isEmpty() -> "∞"
```
Comment (yairm210) [inline core/src/com/unciv/logic/civilization/TechManager.kt:117]: I don't think we need this line anymore if we check the tech directly

File: core/src/com/unciv/ui/victoryscreen/VictoryScreen.kt [hunk 2]
```diff
@@ -76,15 +76,15 @@ class VictoryScreen(val worldScreen: WorldScreen) : PickerScreen() {
 
 
     private fun wonOrLost(description: String) {
-        
-        val endGameMessage = when (description.getPlaceholderParameters()[0]) {
+        // description will be empty when the player loses - don't `[0]`
+        val endGameMessage = when (description.getPlaceholderParameters().firstOrNull()) {
             // Taking the 0th element is a dirty hack to differentiate between the cases
```
Comment (yairm210) [inline core/src/com/unciv/ui/victoryscreen/VictoryScreen.kt:81]: And this comment is deprecated too

**Comment by SomeTroglodyte** [comment]
Quite right. The first one was "Double safety net" kind of thinking, and "maybe statsForNextTurn is an expensive lazy and I don't wanna check right now".
The interesting thing is how I got this to 'trigger', and I think I've understood now - only happened if the only city is a conquest and _puppeted_ therefore keeping its income from me - or so.

**Pull request merged**